### PR TITLE
[FEAT] - add a public Typescript interface for the minimal required serializer interface

### PR DESCRIPTION
**Pull request by pete-the-pete**
@runspired is this even close? Looking at some other TS interface definitions, is seems like there are a few layers of definitions, in this case DS.Store, DS.Model. Should those be defined as well? 

How should this be tested?

### CODE REVIEW

File: packages/store/addon/-private/ts-interfaces/serializer.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import {
+  JsonApiResource
+} from './record-data-json-api';
+export default interface Serializer {
+  store: any;
+  normalize(typeClass: any, hash: any): any;
+  normalizeResponse(store: any, primaryModelClass: any, payload: JsonApiResource, id: string | number, requestType: string): any;
```
Comment (hjdivad) [inline packages/store/addon/-private/ts-interfaces/serializer.ts]: Why would the minimal serializer need to have both `normalize` and `normalizeResponse`?

**Comment by pete-the-pete** [comment]
I don't know. I went off the interface of
https://api.emberjs.com/ember-data/3.10/classes/DS.Serializer

File: packages/store/addon/-private/ts-interfaces/serializer.ts (hunk 1 above)
Comment (hjdivad) [reply packages/store/addon/-private/ts-interfaces/serializer.ts]: @runspired @igorT seems like a good opportunity to nix `normalize`?

Comment (runspired) [reply packages/store/addon/-private/ts-interfaces/serializer.ts]: `Store.normalize` is our only non-serializer code that calls `Serializer.normalize`, agreed that we should nix it, would involve deprecating `Store.normalize`.

**Comment by runspired** [comment]
We may also have to include `serializeIntoHash` due to their usage in in `JSONAPIAdapter` and `RESTAdapter` for `updateRecord` and `createRecord`. However given these are user configurable hooks it may be viable to simply make a note that using these adapters out-of-the-box with a custom serializer will require a tiny amount of customization as well.

File: packages/store/addon/-private/ts-interfaces/serializer.ts (hunk 1 above)
Comment (pete-the-pete) [reply packages/store/addon/-private/ts-interfaces/serializer.ts]: Does that mean I should remove it from this? or add the deprecation to Store.normalize?

Comment (hjdivad) [reply packages/store/addon/-private/ts-interfaces/serializer.ts]: My view is that we should nix this; the serializer interface should have only `normalizeResponse` and `serialize`

@runspired is quite right that we should separately deprecate the other hooks.

We should add tests to verify that you *can* write a userland serializer that implements only those two hooks.

**Comment by runspired** [comment]
Closing in favor of #6451

**Pull request closed without merging**